lib/child_process.js

```diff
@@ -338,8 +350,21 @@ exports.execFile = function(file /*, args, options, callback*/) {
   return child;
 };
 
-Object.defineProperty(exports.execFile, customPromisifyArgs,
-                      { value: ['stdout', 'stderr'], enumerable: false });
+Object.defineProperty(exports.execFile, util.promisify.custom, {
+  enumerable: false, value: function(...args) {
+    const promise = createPromise();
+    exports.execFile.call(this, ...args, (err, stdout, stderr) => {
+      if (err !== null) {
+        err.stdout = stdout;
+        err.stderr = stderr;
+        promiseReject(promise, err);
+      } else {
+        promiseResolve(promise, { stdout, stderr });
+      }
+    });
+    return promise;
+  }
+});
```

Comment:
These two functions are nearly identical. It should be okay, but it shouldn’t be too hard to merge these, right?